freespace2/freespace.cpp

```diff
@@ -3629,17 +3628,13 @@ void game_maybe_do_dead_popup(float frametime)
 				break;
 
 			// this should only happen during a red alert mission
-			case 3:				
+			case 3:			
+				Assertion(The_mission.flags[Mission::Mission_Flags::Red_alert], "FSO tried to reach a Red_alert state without being in a Red_alert mission.");
```

Comment:
This is not reaching a red-alert state; this is choosing the previous-mission option of the pop-up dialog while in a red-alert mission.